From ee4810cad5f4a05f345fce407ecfd574f4cfc9cd Mon Sep 17 00:00:00 2001
From: Jonas Pfenniger <jonas@pfenniger.name>
Date: Sun, 27 Dec 2009 01:07:16 +0100
Subject: [PATCH 1/2] [http] ServerResponse.prototype.sendHeader change

Now accepting a two-sized array to choose your own reason string.

(Needed by WebSocket, apparently the reasons is not free to choose)
---
 lib/http.js |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/lib/http.js b/lib/http.js
index abb472e..d14d546 100644
--- a/lib/http.js
+++ b/lib/http.js
@@ -261,7 +261,14 @@ sys.inherits(ServerResponse, OutgoingMessage);
 exports.ServerResponse = ServerResponse;
 
 ServerResponse.prototype.sendHeader = function (statusCode, headers) {
-  var reason = STATUS_CODES[statusCode] || "unknown";
+  var reason;
+  if (statusCode.splice) {
+    reason = statusCode[1];
+    statusCode = statusCode[0];
+  } else {
+    reason = STATUS_CODES[statusCode] || "unknown";
+  }
+  
   var status_line = "HTTP/1.1 " + statusCode.toString() + " " + reason + CRLF;
   this.sendHeaderLines(status_line, headers);
 };
-- 
1.6.6

